admin_panel/exam_batch/model.py

```diff
@@ -1,10 +1,46 @@
 from django.db import models
 import uuid
+from utils.timestamp_mixin import TimestampMixin
 
 
-class ExamBatch(models.Model):
+class ExamMode(models.TextChoices):
+    CENTER_BASED = "CENTER_BASED", "Center Based Test"
+    INTERNET_BASED = "INTERNET_BASED", "Internet Based Test"
+    INTERNET_BASED_HYBRID = "INTERNET_BASED_HYBRID", "Internet Based Test (Hybrid)"
+    PAPER_BASED_OSM = "PAPER_BASED_OSM", "Paper Based Test (OSM)"
+    PAPER_BASED_OMR = "PAPER_BASED_OMR", "Paper Based Test (OMR)"
+    PAPER_BASED_ATTACHMENT = "PAPER_BASED_ATTACHMENT", "Paper Based Test (Attachment Type)"
+
+
+class AuthenticationType(models.TextChoices):
+    AADHAR = "AADHAR", "Aadhar Card"
+    PAN = "PAN", "PAN Card"
+    VOTER_ID = "VOTER_ID", "Voter ID"
+    DRIVING_LICENSE = "DRIVING_LICENSE", "Driving License"
+
+
+class CriteriaType(models.TextChoices):
+    OVERALL = "OVERALL", "Overall"
+    SECTION = "SECTION", "Section"
+    SUBJECT = "SUBJECT", "Subject"
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider data migration for date fields</b></div>
<div id="fix">

The removal of nullable `start_date` and `end_date` fields while keeping non-nullable versions at lines 71-72 could cause issues with existing data migrations. Consider keeping the fields nullable or adding a data migration plan.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -70,8 +70,8 @@ class ExamBatch(TimestampMixin):
      # Basic schedule
 -    start_date = models.DateField()
 -    end_date = models.DateField()
 +    start_date = models.DateField(null=True, blank=True)
 +    end_date = models.DateField(null=True, blank=True)
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/sleep-404/ExamEvoke/pull/10#issuecomment-2499983195>#fd6475</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged